src/Compilers/CSharp/Test/Semantic/IOperation/IOperationTests_IIsPatternExpression.cs

```diff
@@ -0,0 +1,480 @@
+        }
+
+        [Fact, WorkItem(19927, "https://github.com/dotnet/roslyn/issues/19927")]
+        public void TestIsPatternExpression_ConstantPatternWithConversion()
+        {
+            string source = @"
+using System;
+class X
+{
+    void M()
+    {
+        int? x = 12, y = 12;
+        if (/*<bind>*/x is (int)12.0/*</bind>*/) Console.WriteLine(y);        
+    }
+}
+";
+            string expectedOperationTree = @"
+IIsPatternExpression (OperationKind.IsPatternExpression, Type: System.Boolean) (Syntax: 'x is (int)12.0')
+  Expression: ILocalReferenceExpression: x (OperationKind.LocalReferenceExpression, Type: System.Int32?) (Syntax: 'x')
+  Pattern: IConstantPattern (OperationKind.ConstantPattern) (Syntax: '(int)12.0')
+      Value: IConversionExpression (ConversionKind.CSharp, Explicit) (OperationKind.ConversionExpression, Type: System.Int32, Constant: 12) (Syntax: '(int)12.0')
+          ILiteralExpression (OperationKind.LiteralExpression, Type: System.Double, Constant: 12) (Syntax: '12.0')
+";
+            var expectedDiagnostics = DiagnosticDescription.None;
+
+            VerifyOperationTreeAndDiagnosticsForTest<IsPatternExpressionSyntax>(source, expectedOperationTree, expectedDiagnostics);
+        }
+
+        [Fact, WorkItem(19927, "https://github.com/dotnet/roslyn/issues/19927")]
+        public void TestIsPatternExpression_ConstantPatternWithNoImplicitConversion()
+        {
+            string source = @"
+using System;
+class X
+{
+    void M()
+    {
+        int? x = 12, y = 12;
+        if (/*<bind>*/x is 12.0/*</bind>*/) Console.WriteLine(y);        
+    }
+}
+";
+            string expectedOperationTree = @"
+IIsPatternExpression (OperationKind.IsPatternExpression, Type: System.Boolean) (Syntax: 'x is 12.0')
+  Expression: ILocalReferenceExpression: x (OperationKind.LocalReferenceExpression, Type: System.Int32?) (Syntax: 'x')
+  Pattern: IConstantPattern (OperationKind.ConstantPattern) (Syntax: '12.0')
+      Value: IConversionExpression (ConversionKind.CSharp, Implicit) (OperationKind.ConversionExpression, Type: System.Int32, Constant: 12) (Syntax: '12.0')
+          ILiteralExpression (OperationKind.LiteralExpression, Type: System.Double, Constant: 12) (Syntax: '12.0')
+";
+            var expectedDiagnostics = new DiagnosticDescription[] {
+                // CS0266: Cannot implicitly convert type 'double' to 'int?'. An explicit conversion exists (are you missing a cast?)
+                //         if (/*<bind>*/x is 12.0/*</bind>*/) Console.WriteLine(y);        
+                Diagnostic(ErrorCode.ERR_NoImplicitConvCast, "12.0").WithArguments("double", "int?").WithLocation(8, 28)
+            };
+
+            VerifyOperationTreeAndDiagnosticsForTest<IsPatternExpressionSyntax>(source, expectedOperationTree, expectedDiagnostics);
+        }
+
+        [Fact, WorkItem(19927, "https://github.com/dotnet/roslyn/issues/19927")]
+        public void TestIsPatternExpression_ConstantPatternWithNoConversion()
```

Comment:
There is an implicit conversion here. Can we add a test for something that really shouldn't have a conversion, like just 12, and rename this to reflect the implicit conversion?